tests/testthat/test-sector_profile.R

```diff
@@ -127,3 +134,257 @@ test_that("at company level, two matched and one unmatched products yield `value
   other <- pull(filter(out, !is.na(risk_category)), value)
   expect_equal(sort(other), c(0, 0, 2 / 3))
 })
+
+test_that("at product level, given a `clustered` matching one but not a second `type` of scenario, when the `scenarios` dataset has the two types, then the second `type` and its corresponding `scenario` are still present in `grouped_by` (#739#issuecomment-1977426095)", {
+  # styler: off
+  companies <- tribble(
+    ~companies_id, ~clustered, ~activity_uuid_product_uuid, ~tilt_sector, ~tilt_subsector,       ~type,     ~sector,  ~subsector,
+              "a",        "c",                 "unmatched",          "c",             "c",       "ipr", "land use",   "land use",
+              "a",        "c",                 "unmatched",          "c",             "c",       "weo",         NA,           NA,
+  )
+  scenarios <- tribble(
+       ~sector,   ~subsector,  ~year, ~reductions, ~type, ~scenario,
+    "land use",   "land use",   2050,         0.3, "ipr",       "i",
+       "total",     "energy",   2050,         0.6, "weo",       "w",
+  )
+  # styler: on
+  product <- sector_profile(companies, scenarios) |> unnest_product()
+
+  expect_true("ipr_i_2050" %in% product$grouped_by)
+  expect_true("weo_w_2050" %in% product$grouped_by)
+})
+
+# TODO: Create a constructor for Tilman's example
+
+test_that("at product level, Tilman's example yields what he expects", {
+  # https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064
```

Comment:
For review, it would be great if we know what's special about Tilman's example. right?